src/main/scala/beam/utils/scenario/UrbanSimScenarioLoader.scala

```diff
@@ -229,6 +255,34 @@ class UrbanSimScenarioLoader(
     )
   }
 
+
+  def buildBeamVehicle(
+                        map: Map[Id[BeamVehicleType], BeamVehicleType],
+                        info: VehicleInfo,
+                        randomSeed: Int
+                      ): BeamVehicle = {
+    val matsimVehicleType: VehicleType =
+      VehicleUtils.getFactory.createVehicleType(Id.create(info.vehicleTypeId, classOf[VehicleType]))
+    val matsimVehicle: Vehicle =
+      VehicleUtils.getFactory.createVehicle(Id.createVehicleId(info.vehicleId), matsimVehicleType)
+
+    val beamVehicleId = Id.create(matsimVehicle.getId, classOf[BeamVehicle])
+    val beamVehicleTypeId = Id.create(info.vehicleTypeId, classOf[BeamVehicleType])
+
+    val beamVehicleType = map(beamVehicleTypeId)
+
+    val vehicleManagerId =
+      VehicleManager.createOrGetReservedFor(info.householdId, VehicleManager.TypeEnum.Household).managerId
+    val powerTrain = new Powertrain(beamVehicleType.primaryFuelConsumptionInJoulePerMeter)
+    new BeamVehicle(
+      beamVehicleId,
+      powerTrain,
+      beamVehicleType,
+      new AtomicReference(vehicleManagerId),
+      randomSeed = randomSeed
+    )
+  }
```

Comment:
Using the suggested approach we might be able to get rid of additional code here